gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java

```diff
@@ -1,35 +1,217 @@
+
     private final PatientMigrationRequestDao migrationRequestDao;
+    private final MigrationStatusLogService migrationStatusLogService;
+    private final AttachmentHandlerService attachmentHandlerService;
+    private final PatientAttachmentLogService patientAttachmentLogService;
+    private final XPathService xPathService;
     private final SendACKMessageHandler sendACKMessageHandler;
+    private final SendNACKMessageHandler sendNACKMessageHandler;
 
-    public void handleMessage(InboundMessage inboundMessage, String conversationId) throws JAXBException {
+    public void handleMessage(InboundMessage inboundMessage, String conversationId) throws JAXBException,
+        InlineAttachmentProcessingException {
 
         COPCIN000001UK01Message payload = unmarshallString(inboundMessage.getPayload(), COPCIN000001UK01Message.class);
         PatientMigrationRequest migrationRequest = migrationRequestDao.getMigrationRequest(conversationId);
 
-        sendAckMessage(payload, conversationId, migrationRequest.getLosingPracticeOdsCode());
+        try {
+            Document ebXmlDocument = getEbXmlDocument(inboundMessage);
+            String messageId = xPathService.getNodeValue(ebXmlDocument, MESSAGE_ID_PATH);
+            PatientAttachmentLog patientAttachmentLog = patientAttachmentLogService.findAttachmentLog(messageId, conversationId);
+
+            if (patientAttachmentLog != null) {
+
+                if (isManifestMessage(ebXmlDocument)) {
+                    var fragments = extractFragmentsAndAddRecordToPatientAttachmentLog(migrationRequest, patientAttachmentLog,
+                        conversationId, inboundMessage);
+
+                    List<PatientAttachmentLog> fragmentLogs = patientAttachmentLogService.findAttachmentLogsByParentMid(conversationId,
+                            patientAttachmentLog.getMid()).stream()
+                        .filter(PatientAttachmentLog::getUploaded)
+                        .toList();
+
+                    if (fragments == fragmentLogs.size()) {
+                        mergeFragments(fragmentLogs);
+                    }
+                } else {
+                    storeEhrExtractAttachment(patientAttachmentLog, inboundMessage, conversationId);
+                    patientAttachmentLog.setUploaded(true);
+                    patientAttachmentLogService.updateAttachmentLog(patientAttachmentLog, conversationId);
+
+                    List<PatientAttachmentLog> totalFragmentsList =
+                        patientAttachmentLogService.findAttachmentLogsByParentMid(conversationId,
+                        patientAttachmentLog.getParentMid());
+                    List<PatientAttachmentLog> uploadedFragments = totalFragmentsList.stream()
+                        .filter(PatientAttachmentLog::getUploaded)
+                        .toList();
+
+                    if (totalFragmentsList.size() == uploadedFragments.size()) {
```

Comment:
You can remove this, the later ticket will mange this outside of this processes to maintain a single responsibility